File: dataset_reduzido/avatar/Python/Code/codeforces_609_B.py
def _input ( ) : return map ( int , input ( ).split ( ) )
n , m = _input ( )
lst = list ( _input ( ) )
l = { }
for i in range ( n ) :
    if lst [ i ] not in l :
        l [ lst [ i ] ] = 1
    else : l [ lst [ i ] ] += 1
res = 0
for i in range ( n ) :
    if l [ lst [ i ] ] > 1 :
        res += n - i - l [ lst [ i ] ]
        l [ lst [ i ] ] -= 1
    else :
        res += n - i - 1
print ( res )

File: output/Llama-2-100/avatar/Python/Java/codeforces_609_B.java
public class codeforces_609_B {
    public static void main(String[] args) {
        int n = _input();
        int m = _input();
        List<Integer> lst = Arrays.asList( _input() );
        Map<Integer, Integer> l = new HashMap<>();
        for (int i = 0; i < n; i++) {
            if (!l.containsKey(lst.get(i))) {
                l.put(lst.get(i), 1);
            } else {
                l.put(lst.get(i), l.get(lst.get(i)) + 1);
            }
        }
        int res = 0;
        for (int i = 0; i < n; i++) {
            if (l.get(lst.get(i)) > 1) {
                res += n - i - l.get(lst.get(i));
                l.put(lst.get(i), l.get(lst.get(i)) - 1);
            } else {
                res += n - i - 1;
            }
        }
        System.out.println(res);
    }

    public static int _input() {
        Scanner scanner = new Scanner(System.in);
        return scanner.nextInt() + scanner.nextInt();
    }
}


